### PR TITLE
listener: add metric for connection closed due to network filter config missing

### PR DESCRIPTION
**Pull request by ohadvano**
Additional Description: Follow up to #28477, increasing a metric in cases where a downstream connection is missing a network filter configuration.
Risk Level: low
Testing: Unit test
Docs Changes: Changelog, ECDS docs, listener docs
Release Notes: None
Platform Specific Features: None

### CODE REVIEW

**Review by soulxu — APPROVED** [review]
LGTM, thanks

**Comment by ohadvano** [comment]
@htuch I think that Kuat is still out. Any chance you could take this one?

File: source/extensions/listener_managers/listener_manager/listener_impl.cc [hunk 1]
```diff
@@ -946,7 +946,14 @@ Init::Manager& PerListenerFactoryContextImpl::initManager() { return listener_im
 
 bool ListenerImpl::createNetworkFilterChain(
     Network::Connection& connection, const Filter::NetworkFilterFactoriesList& filter_factories) {
-  return Configuration::FilterChainUtility::buildFilterChain(connection, filter_factories);
+  if (Configuration::FilterChainUtility::buildFilterChain(connection, filter_factories)) {
```
Comment (htuch) [inline source/extensions/listener_managers/listener_manager/listener_impl.cc:949]: Cleaner to make this `if (!...) { ... return false}` and then return true after.

**Review by htuch — CHANGES_REQUESTED** [review]
LGTM modulo tiny nit

File: source/extensions/listener_managers/listener_manager/listener_impl.cc (hunk 1 above)
Comment (ohadvano) [reply source/extensions/listener_managers/listener_manager/listener_impl.cc]: done

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**